octobot_commons/event_tree.py

```diff
@@ -0,0 +1,146 @@
+#  Drakkar-Software OctoBot-Commons
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import asyncio
+from asyncio import Event, ALL_COMPLETED
+
+
+class EventTreeNode(object):
+    __slots__ = ['node_value', 'node_event', 'node_type', 'node_task', 'children']
+
+    def __init__(self, node_value, node_type):
+        self.node_value = node_value
+        self.node_type = node_type
+        self.node_event = Event()
+        self.node_task = None
+        self.children = {}
+
+
+class EventTree(object):
+    __slots__ = ['root']
+
+    def __init__(self):
+        """
+        Init the root node
+        """
+        self.root = EventTreeNode(None, None)
+
+    def set_node(self, value, node_type, node):
+        """
+        Set the node attributes
+        Can raise an exception if the node doesn't exists
+        :param value: the node 'node_value' attribute to set
+        :param node_type: the node 'node_type' attribute to set
+        :param node: the node to update
+        :return: void
+        """
+        self.__set_node(node, value, node_type)
+
+    def set_node_at_path(self, value, node_type, path):
+        """
+        Set the node attributes
+        Creates the node if it doesn't exists
+        :param value: the node 'node_value' attribute to set
+        :param node_type: the node 'node_type' attribute to set
+        :param path: the node path (as a list of string)
```

Comment:
```
:param path: the node path (as a list of string) => what is the value of these strings ? node_value ?
```

Reply:
No it's a list of keys as string. For example you can pass ["my-parent-1", "my-child-1"] as your node path.

Reply:
I mean can you update the doc with info on what to put in this string to navigate through children ?

Reply:
Oh yeah ok i'll update the doc